### PR TITLE
Extend Algolia search parameters

### PR DESCRIPTION
**Pull request by amureki**
Can't make it properly work with dates yet.

### CODE REVIEW

File: sam/contrib/algolia/tools.py [hunk 1]
```diff
@@ -1,23 +1,57 @@
...
+    start_date: str = None,
+    language: str = None,
+    min_age: int = None,
+    max_age: int = None,
+    _context=None,
+) -> str:
     """
     Search the platform for information that matches the given query.
 
+    Translate the query into a German language.
```
Comment (codingjoe) [inline sam/contrib/algolia/tools.py:23]: This line should be added to the function definition in the `pyproject.toml`.

File: sam/contrib/algolia/tools.py [hunk 2]
```diff
@@ -1,23 +1,57 @@
...
 
+from sam import config
 from sam.contrib import algolia
 from sam.utils import logger
 
 
-def search(query: str, _context=None) -> str:
+def search(
+    query: str,
+    start_date: str = None,
```
Comment (codingjoe) [inline sam/contrib/algolia/tools.py:14]:
```suggestion
    start_date: int = None,
```

**Review by codingjoe — COMMENTED** [review]
I don't think we should add filters that are specific to our indexes. Unless we want Sam to search voiio explicitly. That might be cool if a customer sets up a bot like this.

**Comment by amureki** [comment]
> I don't think we should add filters that are specific to our indexes. Unless we want Sam to search voiio explicitly. That might be cool if a customer sets up a bot like this.

yeah, there are more hard-locks on voiio that we need to fix. I'd be happy to change this, but we also need to introduce a proper plugin system.

**Comment by amureki** [comment]
We'll need to unify codebases first and then I will rework this internally.

**Pull request closed without merging**